samples/extensions/open_gl_interop/open_gl_interop.cpp

```diff
@@ -0,0 +1,931 @@
+
+			// The maxLength includes the NULL character
+			std::vector<GLchar> errorLog(maxLength);
+			glGetShaderInfoLog(shader, maxLength, &maxLength, &errorLog[0]);
+			std::string strError;
+			strError.insert(strError.end(), errorLog.begin(), errorLog.end());
+
+			// Provide the infolog in whatever manor you deem best.
+			// Exit with failure.
+			glDeleteShader(shader);        // Don't leak the shader.
+			LOGE("OpenGL: Shader compilation failed", strError.c_str());
+		}
+		return shader;
+	}
+
+	static GLuint buildProgram(const char *vertexShaderSource, const char *fragmentShaderSource)
+	{
+		GLuint program = glCreateProgram();
+		GLuint vs      = loadShader(vertexShaderSource, GL_VERTEX_SHADER);
+		GLuint fs      = loadShader(fragmentShaderSource, GL_FRAGMENT_SHADER);
+		glAttachShader(program, vs);
+		glAttachShader(program, fs);
+		glLinkProgram(program);
+		glDeleteShader(vs);
+		glDeleteShader(fs);
+		return program;
+	}
+
+	static float getTime()
+	{
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
+		timespec now;
+		clock_gettime(CLOCK_MONOTONIC, &now);
+		float result = now.tv_sec + float(now.tv_nsec / 1000000LL) / 1000.0f;
+		return result;
+#else
+		return glfwGetTime();
+#endif
+	}
+
+  public:
+	void create(const OpenGLInterop::ShareHandles &handles, uint64_t memorySize)
+	{
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
+		EGLint eglMajVers{0}, eglMinVers{0};
+		eglDisp = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+		eglInitialize(eglDisp, &eglMajVers, &eglMinVers);
+		LOGD("EGL init with version %d.%d", eglMajVers, eglMinVers);
+
+		constexpr EGLint confAttr[] = {
+		    EGL_RENDERABLE_TYPE, EGL_OPENGL_ES2_BIT,
+		    EGL_SURFACE_TYPE, EGL_PBUFFER_BIT,
+		    EGL_RED_SIZE, 8,
+		    EGL_GREEN_SIZE, 8,
+		    EGL_BLUE_SIZE, 8,
+		    EGL_ALPHA_SIZE, 8,
+		    EGL_NONE};
+
+		EGLint numConfigs;
+		eglChooseConfig(eglDisp, confAttr, &eglConf, 1, &numConfigs);
+
+		// Create a EGL context
+		constexpr EGLint ctxAttr[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
+
+		eglCtx = eglCreateContext(eglDisp, eglConf, EGL_NO_CONTEXT, ctxAttr);
+
+		// Create an offscreen pbuffer surface, and then make it current
+		constexpr EGLint surfaceAttr[] = {EGL_WIDTH, 10, EGL_HEIGHT, 10, EGL_NONE};
+		eglSurface                     = eglCreatePbufferSurface(eglDisp, eglConf, surfaceAttr);
+		eglMakeCurrent(eglDisp, eglSurface, eglSurface, eglCtx);
+		gladLoadGLES2Loader((GLADloadproc) &eglGetProcAddress);
+#else
+		glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 6);
+		glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+		glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, 1);
+		glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE);
+		window = glfwCreateWindow(SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION, "OpenGL Window", nullptr, nullptr);
+		glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+		glfwMakeContextCurrent(window);
+		gladLoadGL();
+#endif
```

Comment:
If you mean a visible OpenGL backed window where the source texture is displayed, I did consider that and used it while debugging, but you run into all sorts of issues related to v-sync.  

Without any special effort, the GL window will be locked to the refresh rate of the device (usually 60 Hz) and because the example is single threaded, that results in the Vulkan output also being locked to that rate or slower.